### PR TITLE
test getitem_full_keys

### PR DESCRIPTION
**Pull request by marqh**
Test to demonstrate a behaviour problem in handling linear_mosaic instances

### CODE REVIEW

**Comment by marqh** [comment]
@pelson I'm a bit puzzled about what should be done to fix this

**Comment by marqh** [comment]
I have pushed a potential fix, which seems to handle my problem, although I'm not sure that I fully understand the logic which I am adapting

@pelson what do you think? would you expect 
_getitem_full_keys to accept arrays in the keys and handle them as I have done here?

**Comment by pelson** [comment]
Seems reasonable. :+1:

File: biggus/tests/test_linear_mosaic.py [hunk 1]
```diff
@@ -219,6 +219,20 @@ def test_ndarray(self):
                   [8, 9, 10, 11, 10, 11, 12, 13, 14]]
         np.testing.assert_array_equal(result, target)
 
+    def test__getitem_full_keys(self):
+        full_keys = (np.array([0, 1, 2, 3]), slice(None, None, None),
+                     slice(None, None, None), slice(None, None, None))
+        tile1 = self._tile((4, 4, 5, 11))
```
Comment (pelson) [inline biggus/tests/test_linear_mosaic.py]: Would you mind reducing this test to 2 tiles, and 2 keys? I think it makes the test look more complex than it is.

**Comment by marqh** [comment]
I think this now meets my requirements for slicing lazy arrays which are linear mosaics.  I have tested two downstream cases would be functional with this change.  Currently they raise an exception:
```
  File "/net/home/h04/itmh/scitools/biggus/biggus/__init__.py", line 548, in __getitem__
    indexed_array = self._getitem_full_keys(keys)
  File "/net/home/h04/itmh/scitools/biggus/biggus/__init__.py", line 1857, in _getitem_full_keys
    if key != slice(None):
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```

**Pull request merged**